src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/Cas2AssessmentEntityFactory.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.factory
+
+import io.github.bluegroundltd.kfactory.Factory
+import io.github.bluegroundltd.kfactory.Yielded
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas2ApplicationEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas2AssessmentEntity
+import java.time.OffsetDateTime
+import java.util.UUID
+
+class Cas2AssessmentEntityFactory : Factory<Cas2AssessmentEntity> {
+  private var id: Yielded<UUID> = { UUID.randomUUID() }
+  private var createdAt: Yielded<OffsetDateTime> = { OffsetDateTime.now() }
+  private var application: Yielded<Cas2ApplicationEntity> = {
+    Cas2ApplicationEntityFactory()
+      .withCreatedByUser(NomisUserEntityFactory().produce())
+      .produce()
+  }
+  private var nacroReferralId: Yielded<String?> = { null }
```

Comment:
I might be wrong but I think we only need to use `Yielded` when we are generating [discrete values for every object](https://github.com/bluegroundltd/kfactory?tab=readme-ov-file#generating-dynamic-values-every-time), when they're the same for every object generated by the factory I think we won't need it.  But haven't checked this.

Reply:
yep, good point! will update